support/cas-server-support-consent-webflow/src/main/java/org/apereo/cas/web/consent/CasConsentReviewController.java

```diff
@@ -0,0 +1,144 @@
+    }
+    
+    /**
+     * Show consent decisions.
+     *
+     * @param request  the request
+     * @param response the response
+     * @return the view where json data will be rendered
+     */
+    @GetMapping
+    public String showConsent(final HttpServletRequest request,
+                                        final HttpServletResponse response) {
+        return "casConsentReviewView";
+    }
+
+    /**
+     * Endpoint for getting consent decisions in JSON format.
+     *
+     * @param request  the request
+     * @param response the response
+     * @return the consent decisions
+     */
+    @GetMapping("/getConsentDecisions")
+    @ResponseBody
+    public Collection<Map<String, Object>> getConsentDecisions(final HttpServletRequest request,
+                                                            final HttpServletResponse response) {
+        final String principal = WebUtils.getPac4jAuthenticatedUsername();
+        if (!PrincipalResolver.UNKNOWN_USER.equals(principal)) {
+            LOGGER.debug("Fetching consent decisions for principal [{}]", principal);
+            final Collection<ConsentDecision> consentDecisions = this.consentRepository.findConsentDecisions(principal);
+            LOGGER.debug("Resolved consent decisions for principal [{}]: {}", principal, consentDecisions);
+            final Collection<Map<String, Object>> result = new HashSet<>();
+            consentDecisions.stream().forEach(d -> {
+                final Map<String, Object> map = new HashMap<>();
+                map.put("decision", d);
+                map.put("attributes", this.consentEngine.resolveConsentableAttributesFrom(d));
+                result.add(map);
+            });
+            return result;
+        }
+        return null;
+    }
+    
+    /**
+     * Endpoint for deleting single consent decisions.
+     * 
+     * @param decisionId the decision id
+     * @return true / false
+     */
+    @PostMapping("/deleteConsentDecision")
+    @ResponseBody
+    public boolean deleteConsentDecision(@RequestParam final Long decisionId) {
+        final String principal = WebUtils.getPac4jAuthenticatedUsername();
+        LOGGER.debug("Deleting consent decision with id [{}] for principal [{}].", decisionId, principal);        
+        return this.consentRepository.deleteConsentDecision(decisionId, principal);
+    }
+    
+    /**
+     * Endpoint for local logout, no SLO.
+     * 
+     * @param request the request
+     * @param session the session
+     * @return the logout view
+     */
+    @GetMapping("/logout")
+    public String logout(final HttpServletRequest request, final HttpSession session) {
+        LOGGER.debug("Invalidating application session...");
+        session.invalidate();
+        return "casConsentLogoutView";
+    }
+    
+    /**
+     * Endpoint for Cas Client Callback.
+     * 
+     * @param request the request
+     * @param response the response
+     */
+    @GetMapping("/callback")
```

Comment:
`/callback` is also a little too generic. Make sure the endpoint corresponds to the functionality at hand. `/consentReviewCallback` is clearer, for instance.

Reply:
Of course, if this is scoped to the parent controller, such that it turns out to be `/consent/callback` or some such then it's fine and you just need to ensure the parent controller has a more explicit mapping.

Reply:
It has become `/consentReview/callback`